rcon/commands.py

```diff
@@ -300,37 +372,39 @@ def get_autobalance_enabled(self):
         return self._get("autobalanceenabled", can_fail=False)
 
     @_auto_retry
-    def get_logs(self, since_min_ago, filter_=""):
+    def get_logs(self, since_min_ago, filter_="", conn: HLLConnection = None):
+        assert conn is not None
         res = self._request(f"showlog {since_min_ago}")
         if res == "EMPTY":
             return ""
-        self.conn.lock()
+        conn.lock()
         try:
             for i in range(30):
                 if res[-1] == "\n":
                     break
                 try:
-                    res += self.conn.receive(unlock=False).decode()
+                    res += conn.receive(unlock=False).decode()
                 except (
-                    RuntimeError,
-                    BrokenPipeError,
-                    socket.timeout,
-                    ConnectionResetError,
-                    UnicodeDecodeError,
+                        RuntimeError,
+                        BrokenPipeError,
+                        socket.timeout,
+                        ConnectionResetError,
+                        UnicodeDecodeError,
                 ):
                     logger.exception("Failed request")
                     raise HLLServerError(f"showlog {since_min_ago}")
         finally:
-            self.conn.unlock()
+            conn.unlock()
         return res
 
     def get_timed_logs(self, since_min_ago, filter_=""):
-        res = self._timed_request(f"showlog {since_min_ago}")
-        for i in range(30):
-            if res["result"][-1] == "\n":
-                break
-            res["result"] += self.conn.receive().decode()
-        return res
+        with self.with_connection() as conn:
+            res = self._timed_request(f"showlog {since_min_ago}", conn=conn)
+            for i in range(30):
+                if res["result"][-1] == "\n":
+                    break
+                res["result"] += conn.receive().decode()
+            return res
```

Comment:
I don't think this method is used but it's not locking / unlocking its connection

Reply:
It did not do that before, either 🤔 Locking/Unlocking on the connection _itself_ should anyway be pretty useless here. The pool ensures that a connection is not used for another RCon command before the current user of the connection finishes somehow.